### PR TITLE
[Python] Support `Optional[...]` in DuckDBPyType

### PR DESCRIPTION
**Pull request by Tishj**
Often type annotations in python contain `Optional` to indicate that the object can be None.
Types in DuckDB are nullable by default, so we just strip out the NoneTypes introduced by Optional.

The purpose of this PR is just to make the DuckDBPyType conversion more complete

### CODE REVIEW

File: tools/pythonpkg/tests/fast/test_type.py [hunk 1]
```diff
@@ -186,3 +187,19 @@ def test_attribute_accessor(self):
 
         child_type = type.v2.child
         assert str(child_type) == 'MAP(BLOB, BIT)'
+
+    def test_optional(self):
```
Comment (Mause) [inline tools/pythonpkg/tests/fast/test_type.py:191]: Can you add a test with `Union[str, None]` or similar? Newer Python versions encourage `str | None` which is syntax sugar for that

Comment (Tishj) [reply tools/pythonpkg/tests/fast/test_type.py:191]: Thanks, added the test case 👍

**Comment by Tishj** [comment]
Failure is unrelated

Related to a failure caused by PyArrow 13.0.0 that we are aware of

**Comment by Mytherin** [comment]
Can you merge with this main?

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**